swig/include/python/docs/gdal_mdm_docs.i

```diff
@@ -1,3 +1,71 @@
+%feature("docstring")  CreateAttribute "
+
+Create an attribute within a :py:class:`osgeo.gdal.MDArray` or :py:class:`osgeo.gdal.Group`.
+
+See :cpp:func:`GDALIHasAttribute::CreateAttribute`.
+
+Parameters
+----------
+name : str
+    name
+dimensions : list
+    List of dimensions, ordered from the slowest varying
+    dimension first to the fastest varying dimension last.
+    Might be empty for a scalar array (if supported by driver)
+data_type: :py:class:`osgeo.gdal.ExtendedDataType`
+    Attribute data type
+options: dict/list
+    an optional dict or list of driver specific ``NAME=VALUE`` option strings.
+
+Returns
+-------
+
+Attribute:
+    the new :py:class:`osgeo.gdal.Attribute` or ``None`` on failure.
+
+Examples
+--------
+>>> numeric_attr = ar.CreateAttribute('numeric_attr', [], gdal.ExtendedDataType.Create(gdal.GDT_Float64))
```

Comment:
I think it would be good for the examples to be runnable from scratch (except for module imports) and this would be a requirement in any case for #9949